superset-frontend/plugins/plugin-chart-echarts/src/utils/series.ts

```diff
@@ -229,8 +229,10 @@ export function sortRows(
     }
 
     const value =
-      xAxisSortSeries === SortSeriesType.Name && typeof sortKey === 'string'
-        ? sortKey.toLowerCase()
+      xAxisSortSeries === SortSeriesType.Name
+        ? typeof sortKey === 'string'
+          ? sortKey.toLowerCase()
+          : sortKey
```

Comment:
Without this change, sorting by "category name" wouldn't work if the type is numeric